src/graphics/vulkan.zig

```diff
@@ -345,3 +351,139 @@ fn createLogicalDevice() void {
+				.height = @min(self.capabilities.maxImageExtent.height, @max(self.capabilities.minImageExtent.height, @max(0, height))),
+			};
+		}
+	};
+
+	fn createImageView(image: c.VkImage) c.VkImageView {
+		const createInfo: c.VkImageViewCreateInfo = .{
+			.sType = c.VK_STRUCTURE_TYPE_IMAGE_VIEW_CREATE_INFO,
+			.image = image,
+			.viewType = c.VK_IMAGE_VIEW_TYPE_2D,
+			.format = imageFormat,
+			.components = .{
+				.a = c.VK_COMPONENT_SWIZZLE_IDENTITY,
+				.r = c.VK_COMPONENT_SWIZZLE_IDENTITY,
+				.g = c.VK_COMPONENT_SWIZZLE_IDENTITY,
+				.b = c.VK_COMPONENT_SWIZZLE_IDENTITY,
+			},
+			.subresourceRange = .{
+				.aspectMask = c.VK_IMAGE_ASPECT_COLOR_BIT,
+				.baseMipLevel = 0,
+				.levelCount = 1,
+				.baseArrayLayer = 0,
+				.layerCount = 1,
+			},
+		};
+		var result: c.VkImageView = undefined;
+		checkResult(c.vkCreateImageView(device, &createInfo, null, &result));
+		return result;
+	}
+
+	fn init() void {
+		const support = SupportDetails.init(main.stackAllocator, physicalDevice);
+		defer support.deinit(main.stackAllocator);
+
+		const surfaceFormat = support.chooseFormat();
+		imageFormat = surfaceFormat.format;
+		const presentMode = support.chooseSwapPresentMode();
+		extent = support.chooseSwapExtent();
+		const imageCount = @min(support.capabilities.minImageCount + 1, support.capabilities.maxImageCount -% 1 +| 1);
+
+		var createInfo: c.VkSwapchainCreateInfoKHR = .{
+			.sType = c.VK_STRUCTURE_TYPE_SWAPCHAIN_CREATE_INFO_KHR,
+			.surface = surface,
+			.minImageCount = imageCount,
+			.imageFormat = surfaceFormat.format,
+			.imageColorSpace = surfaceFormat.colorSpace,
+			.imageExtent = extent,
+			.imageArrayLayers = 1,
+			.imageUsage = c.VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT,
+			.preTransform = support.capabilities.currentTransform,
+			.compositeAlpha = c.VK_COMPOSITE_ALPHA_OPAQUE_BIT_KHR,
+			.presentMode = presentMode,
+			.clipped = c.VK_TRUE,
+			.oldSwapchain = null,
+		};
+		const queueFamilies = findQueueFamilies(physicalDevice);
+		if(queueFamilies.graphicsFamily.? != queueFamilies.presentFamily.?) {
+			const queueFamilyIndices = [_]u32{queueFamilies.graphicsFamily.?, queueFamilies.presentFamily.?};
+			createInfo.imageSharingMode = c.VK_SHARING_MODE_CONCURRENT;
+			createInfo.queueFamilyIndexCount = @intCast(queueFamilyIndices.len);
+			createInfo.pQueueFamilyIndices = &queueFamilyIndices;
+		} else {
+			createInfo.imageSharingMode = c.VK_SHARING_MODE_EXCLUSIVE;
+		}
+
+		checkResult(c.vkCreateSwapchainKHR(device, &createInfo, null, &swapChain));
+		images = main.globalAllocator.alloc(c.VkImage, imageCount);
+		var newImageCount = imageCount;
+		checkResult(c.vkGetSwapchainImagesKHR(device, swapChain, &newImageCount, images.ptr));
+		std.debug.assert(newImageCount == imageCount);
+
+		imageViews = main.globalAllocator.alloc(c.VkImageView, imageCount);
+		for(0..images.len) |i| {
+			imageViews[i] = createImageView(images[i]);
+		}
+	}
+
+	fn deinit() void {
```

Comment:
Idk if it matters, hopefully not, but what is the planned path to re-create surface images on resize? We just scrap whole swap chain?